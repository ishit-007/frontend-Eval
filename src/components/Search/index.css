.search {
  margin-top: 3rem;
  margin-right: 5%;
  display: flex;
  gap: 2px;
  justify-content: flex-end;
}
.search input {
  border: 2px solid black;
  border-radius: 5px;
}
.search i {
  padding-top: 2px;
}
