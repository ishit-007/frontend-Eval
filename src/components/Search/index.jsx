import './index.css';

import React, { useContext } from 'react';

import ProptTypes from 'prop-types';
import { ThemeContext } from '../../contexts/ThemeContext';

const Search = props => {
  const { appTheme } = useContext(ThemeContext);
  const inputChangeHandler = event => {
    props.setSearchedText(event.target.value);
  };
  return (
    <div className="search">
      <input type="text" placeholder="EVENT NAME" onChange={inputChangeHandler} style={{ borderColor: appTheme }} />
      <i className="fa fa-search" style={{ fil: appTheme }}></i>
    </div>
  );
};

Search.propTypes = {
  setSearchedText: ProptTypes.func,
  searchedText: ProptTypes.string,
};
export default Search;
