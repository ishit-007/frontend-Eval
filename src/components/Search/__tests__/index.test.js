import Search from '..';
import renderer from 'react-test-renderer';
import React from 'react';

describe('Search', () => {
  it('should render the Search Component', async () => {
    const tree = renderer.create(<Search />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
