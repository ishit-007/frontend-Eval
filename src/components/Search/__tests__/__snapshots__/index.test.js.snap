// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render the Search Component 1`] = `
<div
  className="search"
>
  <input
    onChange={[Function]}
    placeholder="EVENT NAME"
    type="text"
  />
  <i
    className="fa fa-search"
  />
</div>
`;
